<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GPACalc</class>
 <widget class="QWidget" name="GPACalc">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>东南大学绩点计算器</string>
  </property>
  <property name="styleSheet">
   <string notr="true">font: 11pt &quot;Microsoft YaHei UI&quot;;</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTableView" name="gpaViewer">
     <property name="selectionMode">
      <enum>QAbstractItemView::NoSelection</enum>
     </property>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="horizontalWidget_2" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="bottomMargin">
       <number>20</number>
      </property>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>计算学分：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="totalXF">
        <property name="text">
         <string>0.0</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>平均绩点（4.8制）：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="averageGPA">
        <property name="text">
         <string>0.0000</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>平均分：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="averageScore">
        <property name="text">
         <string>0.0000</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0,0,0">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QPushButton" name="selectAll">
       <property name="text">
        <string>全选</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="selectInv">
       <property name="text">
        <string>反选</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="selectContemp">
       <property name="text">
        <string>选择所有必修课</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="clear">
       <property name="text">
        <string>清空选择</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="loadBtn">
       <property name="text">
        <string>加载成绩</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="export2CSV">
       <property name="text">
        <string>导出成绩为CSV</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>loadBtn</sender>
   <signal>clicked()</signal>
   <receiver>GPACalc</receiver>
   <slot>gpaLoader()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>284</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>gpaLoader()</slot>
 </slots>
</ui>
